modules/rosa-sts-account-roles-terraform.adoc

```diff
@@ -0,0 +1,212 @@
+ifeval::["{context}" == "rosa-understanding-terraform"]
+:tf-full:
+endif::[]
+:_content-type: PROCEDURE
+
+[id="sd-terraform-account-roles_{context}"]
+ifdef::tf-full[]
+= Account roles Terraform example
+endif::tf-full[]
+ifndef::tf-full[]
+= Creating your account-wide IAM roles with Terraform
+endif::tf-full[]
+
+The following examples shows how Terraform can be used to create your AWS IAM account roles for ROSA.
+
+.Procedure
+
+. Check your AWS account for existing roles and policies:
++
+[source,terminal]
+----
+$ rosa list account-roles
+----
+
+. Create the Terraform files locally by using the following codes samples:
+.. Copy this template into a file  named `main.tf`. Running the main.tf file calls the Red Hat Cloud Services Terraform provider, which allows you to use OpenShift services with Terraform.
++
+[source,terraform]
+----
+#
+# Copyright (c) 2022 Red Hat, Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+terraform {
+  required_providers {
+    aws = {
+      source  = "hashicorp/aws"
+      version = ">= 4.20.0"
+    }
+    rhcs = {
+      version = ">= 1.1.0"
+      source  = "terraform-redhat/rhcs"
+    }
+  }
+}
+
+provider "rhcs" {
+  token = var.token
+  url   = var.url
+}
```

Comment:
Please remove this "rhcs" provider section and add instructions on how to set this as environment variable
User should export the environment variable:
export OCM_TOKEN="<the token>"
This way is safer
External users shouldn't set the url, they should use default value (production)